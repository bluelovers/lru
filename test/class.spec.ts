/**
 * Created by user on 2020/4/11.
 */

import LRU from '../index';
import { EventEmitter } from "events";
import QuickLRU from 'quick-lru';

describe('inherits', () =>
{
	describe('EventEmitter', () =>
	{

		Object.getOwnPropertyNames(EventEmitter.prototype)
			.forEach(prop =>
			{

				if (typeof EventEmitter.prototype[prop] !== 'function')
				{
					return
				}

				it(prop, () =>
				{

					expect(LRU.prototype).toHaveProperty(prop)
					expect(typeof LRU.prototype[prop]).toStrictEqual(typeof EventEmitter.prototype[prop])

				});

			})
		;

	});

	describe('QuickLRU', () =>
	{

		Object.getOwnPropertyNames(QuickLRU.prototype)
			.forEach(prop =>
			{

				if (prop === 'size') return;

				it(prop, () =>
				{

					expect(LRU.prototype).toHaveProperty(prop)
					expect(typeof LRU.prototype[prop]).toStrictEqual(typeof QuickLRU.prototype[prop])

				});

			})
		;

	});

	describe('Instance', () =>
	{
		const lru = new LRU();

		([
			['size', 'number'],
			['length', 'number'],
		] as ([keyof LRU<any, any>, string])[])
			.forEach(([prop, type]) =>
			{
				it(prop, () =>
				{
					expect(lru).toHaveProperty(prop)
					expect(typeof lru[prop]).toStrictEqual(type)
				})
			})
		;

		it.skip('toMatchSnapshot', () => {
			expect(lru).toMatchSnapshot();
		})

	});

	it.skip('toMatchSnapshot', () => {
		expect(LRU).toMatchSnapshot();
		expect(LRU.prototype).toMatchSnapshot();
		expect(Object.getOwnPropertyNames(LRU.prototype)).toMatchSnapshot();
		expect(Object.getOwnPropertyNames(QuickLRU.prototype)).toMatchSnapshot();
		expect(Object.getOwnPropertyNames(EventEmitter.prototype)).toMatchSnapshot();
	})

});
