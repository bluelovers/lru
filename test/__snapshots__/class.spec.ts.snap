// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inherits Instance toMatchSnapshot 1`] = `
LRU {
  "_events": Object {},
  "_eventsCount": 0,
  "_maxListeners": undefined,
  "_size": 0,
  "cache": Map {},
  "maxSize": 1000,
  "oldCache": Map {},
  "onEviction": [Function],
  Symbol(kCapture): false,
}
`;

exports[`inherits toMatchSnapshot 1`] = `[Function]`;

exports[`inherits toMatchSnapshot 2`] = `
LRU {
  "addListener": [Function],
  "emit": [Function],
  "eventNames": [Function],
  "getMaxListeners": [Function],
  "listenerCount": [Function],
  "listeners": [Function],
  "off": [Function],
  "on": [Function],
  "once": [Function],
  "prependListener": [Function],
  "prependOnceListener": [Function],
  "rawListeners": [Function],
  "removeAllListeners": [Function],
  "removeListener": [Function],
  "setMaxListeners": [Function],
}
`;

exports[`inherits toMatchSnapshot 3`] = `
Array [
  "constructor",
  "remove",
  "length",
  "setMaxListeners",
  "getMaxListeners",
  "emit",
  "addListener",
  "on",
  "prependListener",
  "once",
  "prependOnceListener",
  "removeListener",
  "off",
  "removeAllListeners",
  "listeners",
  "rawListeners",
  "listenerCount",
  "eventNames",
]
`;

exports[`inherits toMatchSnapshot 4`] = `
Array [
  "constructor",
  "_set",
  "get",
  "set",
  "has",
  "peek",
  "delete",
  "clear",
  "keys",
  "values",
  "size",
]
`;

exports[`inherits toMatchSnapshot 5`] = `
Array [
  "constructor",
  "_events",
  "_eventsCount",
  "_maxListeners",
  "setMaxListeners",
  "getMaxListeners",
  "emit",
  "addListener",
  "on",
  "prependListener",
  "once",
  "prependOnceListener",
  "removeListener",
  "off",
  "removeAllListeners",
  "listeners",
  "rawListeners",
  "listenerCount",
  "eventNames",
]
`;
